### PR TITLE
Mrc 2564 Move survey and program (SAP) UI component to baseline step

### PR DESCRIPTION
**Pull request by carbowizzy**
This PR moves input upload to baseline, including corresponding actions and tests.

## Type of version change
_Delete as appropriate_

Minor

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue [hunk 1]
```diff
@@ -85,9 +53,9 @@
 
 <script lang="ts">
     import Vue from "vue";
-    import Treeselect from '@riophae/vue-treeselect';
+    import TreeSelect from '@riophae/vue-treeselect';
```
Comment (mwoodbri) [inline src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue:56]: I was quite surprised this worked - but turns out the package does both a default and named export. We seem to use both `Treeselect` and `TreeSelect` at the moment - I have no preference.

File: src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue [hunk 2]
```diff
@@ -156,26 +122,14 @@
                     return surveyAndProgram.selectedDataType != null;
                 },
                 anc: ({surveyAndProgram}: {surveyAndProgram: SurveyAndProgramState}) => ({
-                    valid: !!surveyAndProgram.anc,
-                    fromADR: !!surveyAndProgram.anc?.fromADR,
-                    error: surveyAndProgram.ancError,
-                    existingFileName: (surveyAndProgram.anc && surveyAndProgram.anc.filename)|| surveyAndProgram.ancErroredFile,
                     available: !surveyAndProgram.ancError && surveyAndProgram.anc
-                } as PartialFileUploadProps),
+                } as any),
```
Comment (mwoodbri) [inline src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue:126]: Not sure you even need the `as any` here - given there's no need to explicitly broaden the type

**Review by mwoodbri — COMMENTED** [review]
Obviously this is a transitional step so not the easiest thing to review, but it seems to roughly have a net zero effect on line count!

Do you have any further guidance on manual testing i.e. whether it makes sense at all and if so whether the app should still be fully functioning? Superficially it is but I guess some of the validation won't be, looking at the next ticket to be implemented?

File: src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue (hunk 2 above)
Comment (carbowizzy) [reply src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue]: You are right, any removed.

File: src/app/static/src/tests/components/surveyAndProgram/fileUploads.ts [hunk 3]
```diff
@@ -110,25 +124,25 @@ export function testUploadComponent(name: string, position: number) {
 
     it(`${name} upload is valid if data is present`, () => {
         const store = createSut(successState);
-        const wrapper = shallowMount(SurveyAndProgram, {store, localVue});
+        const wrapper = shallowMount(Baseline, {store, localVue});
```
Comment (EmmaLRussell) [inline src/app/static/src/tests/components/surveyAndProgram/fileUploads.ts:127]: These tests should really move into the `components/baseline` folder to test functionality which is now happening from the `Baseline` component

File: src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue (hunk 1 above)
Comment (EmmaLRussell) [reply src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue:56]: Me neither really - `TreeSelect` is a little easier to read

**Review by EmmaLRussell — COMMENTED** [review]
LGTM, behaves how I would expect it to in this intermediate stage.

File: src/app/static/src/tests/components/surveyAndProgram/fileUploads.ts (hunk 3 above)
Comment (carbowizzy) [reply src/app/static/src/tests/components/surveyAndProgram/fileUploads.ts:127]: done

File: src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue [hunk 4]
```diff
@@ -9,47 +9,15 @@
         </div>
```
Comment (Nicolas-Dolan) [inline src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue:9]: Silly question, but is this component eventually going to be discontinued/replaced or will it still serve some function?

Comment (carbowizzy) [reply src/app/static/src/app/components/surveyAndProgram/SurveyAndProgram.vue:9]: I will serve as Review inputs where the charts will be displayed.

**Review by EmmaLRussell — APPROVED** [review]
Thanks!

**Review by Nicolas-Dolan — APPROVED** [review]

**Review by mwoodbri — APPROVED** [review]

**Pull request merged**